File: src/main/java/org/kite9/diagram/model/position/RenderingInformation.java
package org.kite9.diagram.model.position;

/**
 * This holds formatting information for the graphical renderer to use.
 * 
 * @author robmoffat
 *
 */
public interface RenderingInformation {
	
	/**
	 * Returns true if this item should be drawn
	 */
	public boolean isRendered();

	public abstract void setRendered(boolean r);

	/**
	 * Returns the bounds consumed by this element
	 */
	public Dimension2D getSize();
	
	/**
	 * Returns top-left most coordinate of element.
	 */
	public Dimension2D getPosition();

}

File: kite9-visualization/src/commonMain/kotlin/org/kite9/diagram/model/position/RenderingInformation.kt
package org.kite9.diagram.model.position

/**
 * This holds formatting information for the graphical renderer to use.
 *
 * @author robmoffat
 */
interface RenderingInformation {
    /**
     * Returns true if this item should be drawn
     */
    var rendered: Boolean

    /**
     * Returns the bounds consumed by this element
     */
    var size: Dimension2D?

    /**
     * Returns top-left most coordinate of element.
     */
    var position: Dimension2D?
}